completions/feh

```diff
@@ -30,11 +30,13 @@ _comp_cmd_feh()
             local font_path
             # font_path="$(imlib2-config --prefix 2>/dev/null)/share/imlib2/data/fonts"
             # _comp_compgen -C "$font_path" -- -f -X "!*.@([tT][tT][fF])" -S /
-            for ((i = ${#words[@]} - 1; i > 0; i--)); do
+            for ((i = ${#words[@]} - 2; i > 0; i--)); do
                 if [[ ${words[i]} == -@(C|-fontpath) ]]; then
                     font_path="${words[i + 1]}"
-                    _comp_compgen -aC "$font_path" -- \
-                        -f -X "!*.@([tT][tT][fF])" -S /
+                    if [[ -d $font_path ]]; then
+                        _comp_compgen -aC "$font_path" -- \
+                            -f -X "!*.@([tT][tT][fF])" -S /
```

Comment:
Unrelated to the present topic, but I'm wondering the reason that we use `-f -X "!globpat"` instead of simply `-G "globpat"`. In the codebase, we only have two uses of `-G globpat`:

```
./completions/mdtool:35:                _comp_compgen -- -o filenames -G"*.mds"
./completions/rcs:24:    _comp_compgen -aR -- -G "$dir/$file*,v"
```

while we have many for `-f -X "!globpat"`:

```bash
./completions/explodepkg:3:complete -o plusdirs -f -X '!*.t[bglx]z' explodepkg
./completions/feh:32:            # _comp_compgen -C "$font_path" -- -f -X "!*.@([tT][tT][fF])" -S /
./completions/feh:38:                            -f -X "!*.@([tT][tT][fF])" -S /
./completions/gdb:41:        _comp_compgen -a -- -f -X '!?(*/)core?(.?*)' -o plusdirs
./completions/java:87:            _comp_compgen -av tmp -c "$i/$cur" -- -f -X '!*.class'
./completions/lzip:47:    _comp_compgen -- -f -X "*.lz" -o plusdirs
./completions/sbopkg:67:        _comp_compgen -aC "$QUEUEDIR" -- -f -X "!*.sqf"
./completions/_slackpkg:68:                _comp_compgen -C "$confdir/templates" -- -f -X \
                    "!?*.template" && COMPREPLY=("${COMPREPLY[@]%.template}")
./completions/upgradepkg:18:        _comp_compgen -- -P "$prev%" -f -X "!*.@(t[bgxl]z)" || nofiles=set
```

Reply:
They are introduced in the following commits, where I don't find any discussions about using `-f -X` over `-G`.

- explodepkg, upgradepkg [`c340b7d`](https://github.com/scop/bash-completion/commit/c340b7d8d12e144945facdd8165847e8af215609#diff-a4757074ff650000804fd3eaabe9b0a9e02e33040ca5b8afd4c0275fc5f3e136R3992)
- feh [`c7b9d08`](https://github.com/scop/bash-completion/commit/c7b9d08d171baa782bbb56ff142ea8572eecc778#diff-06e415730b4e63cbeefed1f48b7b7f8718b1a1cb3448cb4065d9e07e04545858R41-R42)
- gdb [`f815736`](https://github.com/scop/bash-completion/commit/f815736ffe110ee079bc9d0f37394cd3c3e8ad33#diff-c78270b40e79398b235e15b9302981a056681f2589ad074d8744cdebf4df1193R33)
- java [`920b1f9`](https://github.com/scop/bash-completion/commit/920b1f93b64b4e95b7097085290ea6f998d8347f#diff-20440e71a6a7ec0be45bf8bec35417781bcdc464f87f1701f577036237621032R78)
- lzip [`f09ced6`](https://github.com/scop/bash-completion/commit/f09ced610d6f9c9ea694d2eade7bdcf75e535afc#diff-87387e55a9536f7ac2aa5fce39d9a0442c29ec2058215563388d5a284a9ac1c1R39)
- sbopkg [`b346174`](https://github.com/scop/bash-completion/commit/b34617469777de05257d86b3810578b3ac4ca97f#diff-01169a17e0b5e9275b383a140c6296f2a032cfc7c89dd214b7dc8ab7cc2a7f10R75)
- _slackpkg [`b346174`](https://github.com/scop/bash-completion/commit/b34617469777de05257d86b3810578b3ac4ca97f#diff-8da8ce6f20c709a0b742c6a50941c01b80baa18d1395c68e476cd14b0d981146R46)